src/types/routes.ts

```diff
@@ -93,328 +93,753 @@ export type H<
-    I7 extends Input = I & I2 & I3 & I4 & I5 & I6,
-    I8 extends Input = I & I2 & I3 & I4 & I5 & I6 & I7,
-    I9 extends Input = I & I2 & I3 & I4 & I5 & I6 & I7 & I8,
-    E2 extends Env = E,
-    E3 extends Env = E,
-    E4 extends Env = E,
-    E5 extends Env = E,
-    E6 extends Env = E,
-    E7 extends Env = E,
-    E8 extends Env = E,
-    E9 extends Env = E,
-    E10 extends Env = IntersectNonAnyTypes<[E, E2, E3, E4, E5, E6, E7, E8, E9]>,
-  >(
-    path: P,
-    middleware: MiddlewareHandler<E2, MergedPath, I>,
-    middleware_2: MiddlewareHandler<E3, MergedPath, I2>,
-    middleware_3: MiddlewareHandler<E4, MergedPath, I3>,
-    middleware_4: MiddlewareHandler<E5, MergedPath, I4>,
-    middleware_5: MiddlewareHandler<E6, MergedPath, I5>,
-    middleware_6: MiddlewareHandler<E7, MergedPath, I6>,
-    middleware_7: MiddlewareHandler<E8, MergedPath, I7>,
-    middleware_8: MiddlewareHandler<E9, MergedPath, I8>,
-    handler: H<E10, MergedPath, I9, R, M>,
-    options?: RouteOptions,
-  ): FrogBase<
-    E,
-    S &
-      ToSchema<M, MergePath<BasePath, P>, I9['in'], MergeTypedResponseData<R>>,
-    BasePath
-  >
-
-  // app.get(path, middleware x9, handler)
-  <
-    P extends string,
-    MergedPath extends MergePath<BasePath, P> = MergePath<BasePath, P>,
-    R extends HandlerResponse<any> = any,
-    I extends Input = BlankInput,
-    I2 extends Input = I,
-    I3 extends Input = I & I2,
-    I4 extends Input = I & I2 & I3,
-    I5 extends Input = I & I2 & I3 & I4,
-    I6 extends Input = I & I2 & I3 & I4 & I5,
-    I7 extends Input = I & I2 & I3 & I4 & I5 & I6,
-    I8 extends Input = I & I2 & I3 & I4 & I5 & I6 & I7,
-    I9 extends Input = I & I2 & I3 & I4 & I5 & I6 & I7 & I8,
-    I10 extends Input = I & I2 & I3 & I4 & I5 & I6 & I7 & I8 & I9,
-    E2 extends Env = E,
-    E3 extends Env = E,
-    E4 extends Env = E,
-    E5 extends Env = E,
-    E6 extends Env = E,
-    E7 extends Env = E,
-    E8 extends Env = E,
-    E9 extends Env = E,
-    E10 extends Env = E,
-    E11 extends Env = IntersectNonAnyTypes<
-      [E, E2, E3, E4, E5, E6, E7, E8, E9, E10]
-    >,
-  >(
-    path: P,
-    middleware: MiddlewareHandler<E2, MergedPath, I>,
-    middleware_2: MiddlewareHandler<E3, MergedPath, I2>,
-    middleware_3: MiddlewareHandler<E4, MergedPath, I3>,
-    middleware_4: MiddlewareHandler<E5, MergedPath, I4>,
-    middleware_5: MiddlewareHandler<E6, MergedPath, I5>,
-    middleware_6: MiddlewareHandler<E7, MergedPath, I6>,
-    middleware_7: MiddlewareHandler<E8, MergedPath, I7>,
-    middleware_8: MiddlewareHandler<E9, MergedPath, I8>,
-    middleware_9: MiddlewareHandler<E10, MergedPath, I9>,
-    handler: H<E11, MergedPath, I10, R, M>,
-    options?: RouteOptions,
-  ): FrogBase<
-    E,
-    S &
-      ToSchema<M, MergePath<BasePath, P>, I10['in'], MergeTypedResponseData<R>>,
-    BasePath
-  >
-}
+> = /* We enforce `options` parameter to not be partial for Cast Actions*/
+M extends 'cast-action'
```

Comment:
Hmm. I would rather avoid this ternary. Can we instead make `options` optional within the parameter definition itself if the method extends `"cast-action"`?

```tsx
(
  path: P,
  handler: H<...>,
  ...rest: (M extends 'cast-action' ? [options: RouteOptions<M>] : [options?: RouteOptions<M>])
): FrogBase<...>
```

Reply:
brilliant

Reply:
done!